pinecone/index_connection.go

```diff
@@ -49,11 +56,88 @@ func newIndexConnection(in newIndexParameters) (*IndexConnection, error) {
 	return &idx, nil
 }
 
+// Close closes the connection to a Pinecone index.
+//
+// Returns a pointer to an IndexConnection object and an error if the connection cannot be closed.
+//
+// Parameters:
+//   idx: The IndexConnection object.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//	  ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//	  SourceTag: "your_source_identifier", // optional
+//	}
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//	if err != nil {
+//		log.Fatalf("Failed to create Client: %v", err)
+//  }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//  if err != nil {
+//	  fmt.Println("Error:", err)
+//  }
+//
+//  idx, err := pc.Index(idxs[0].Host)
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//	  return
+//	}
+//
+//  err = idx.Close()
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//  }
 func (idx *IndexConnection) Close() error {
 	err := idx.grpcConn.Close()
 	return err
 }
 
+// UpsertVectors indexes vectors into a Pinecone index.
+//
+// Parameters:
+//  - ctx: A context.Context object controls the request's lifetime,
+//  allowing for the request to be canceled or to timeout according to the context's deadline.
+//  - in: The vectors to index.
```

Comment:
nit: I think we refer to it as an "upsert" elsewhere rather than an "index": 

https://github.com/pinecone-io/pinecone-ts-client/blob/4fe3e04e05c4c8748262e5a00446bcb656d379eb/src/data/index.ts#L426

https://github.com/pinecone-io/pinecone-python-client/blob/ab3ccfee30577e6a9bcab8d8a8f40b3696e1cda4/pinecone/data/index.py#L149

Reply:
Oops ty!